### PR TITLE
feat: navigate aggregations generated from the query bar ai COMPASS-7106

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

In case AI returns aggregation on the "Documents" tab, we navigate users to the "Aggregations" tab and create a pipeline. We also show a guide cue so users are aware of what just happened.

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-aggregations/src/modules/pipeline-builder/pipeline-ai.ts
Comment (gribnoysup) [inline packages/compass-aggregations/src/modules/pipeline-builder/pipeline-ai.ts:56]: While makes it definitely clearer, we currently have 3 people working on the code in the same area, I'd maybe stay away from renaming for now just to limit the amount of possible git conflicts between PRs

File: packages/compass-aggregations/src/stores/store.ts
Comment (gribnoysup) [inline packages/compass-aggregations/src/stores/store.ts:259]: So I'd suggest either to make the event name more scoped to the use-case you are introducing or event handler should be handling this as a more generic event that can be used to just programmatically open aggregation subtab inside collection tab, optionally indicating somehow that it's coming from query tab AI feature. Currently you are mixing two approaches: the event name is very generic indicating that it can be used to just open aggregation tab, but action that is being dispatched is purely focused on handling this as if it's always tab being opened in relation to AI endpoint returning query instead of aggregation

File: packages/compass-query-bar/src/stores/ai-query-reducer.ts
Comment (gribnoysup) [inline packages/compass-query-bar/src/stores/ai-query-reducer.ts:243]: This type assertion is incorrect, we validate the response before returning it from `getQuery` method, so what you get here will always match expected backend schema (or request will fail already and you will not end up in this part of the action)

File: packages/compass-query-bar/src/stores/ai-query-reducer.ts
Comment (gribnoysup) [inline packages/compass-query-bar/src/stores/ai-query-reducer.ts:269]: Should we check that there is no query returned with aggergation / move this `if (aggregation)` block under the check that comes afterwards that determines that query came back empty? When testing backend locally, I saw sometimes that returned response included both query and aggergation sometimes, something like `{query: { find: '{ _id: 1 }', aggregation: { pipeline: '[{ $match: { _id: 1 } }]' } }`, which is a valid query and we probably don't need to switch in this case?

File: packages/compass-components/src/components/generative-ai/generative-ai-input.tsx
Comment (gribnoysup) [inline packages/compass-components/src/components/generative-ai/generative-ai-input.tsx:157]: The way this is currently implemented, this state doesn't really do anything, you are mixing controlled and uncontrolled components here. If you want component to be controlled, the state of the guide cue should always be read and changed from the outside, not from internal state, if you want it to be uncontrolled, you don't need `onHideGuideCue` and this `showGuideCue` should rather be called something like `showGuideCueOnMount` or `initialShowGuideCue` to indicate that it's not fully controlling the state of the cue component

File: packages/compass-query-bar/src/stores/ai-query-reducer.ts
Comment (alenakhineika) [reply packages/compass-query-bar/src/stores/ai-query-reducer.ts:267]: You are too fast with commenting :) I already changed this locally after rebasing from the main, but haven't pushed yet.

Comment (gribnoysup) [reply packages/compass-query-bar/src/stores/ai-query-reducer.ts:267]: Oh, it's a draft, I didn't notice, sorry! Disregard everything that doesn't make sense with the changes you are still doing then 🙇

File: packages/compass-components/src/components/generative-ai/index.ts
Comment (alenakhineika) [reply packages/compass-components/src/components/generative-ai/index.ts]: Yep.

File: packages/compass-query-bar/src/stores/ai-query-reducer.ts
Comment (alenakhineika) [reply packages/compass-query-bar/src/stores/ai-query-reducer.ts]: Yep gone after rebasing from main and refactoring.

File: packages/compass-query-bar/src/stores/ai-query-reducer.ts
Comment (alenakhineika) [reply packages/compass-query-bar/src/stores/ai-query-reducer.ts]: Pushed.

**Review by gribnoysup — COMMENTED** [review]
Overall already looks good, but I think the ai-input / ai-guide-cue implementation can be simplified and dependency on aggregation store can be completely removed

File: packages/compass-components/src/components/generative-ai/ai-guide-cue.tsx
Comment (alenakhineika) [reply packages/compass-components/src/components/generative-ai/ai-guide-cue.tsx]: So I kept onHideGuideCue and noped setOpen. Let me know how this look now 🙈

File: packages/compass-components/src/components/generative-ai/generative-ai-input.tsx
Comment (alenakhineika) [reply packages/compass-components/src/components/generative-ai/generative-ai-input.tsx]: Refactored as we discussed. Please let me know if missed something 🙏

File: packages/compass-components/src/components/generative-ai/generative-ai-input.tsx
Comment (Anemy) [inline packages/compass-components/src/components/generative-ai/generative-ai-input.tsx:265]: There's a typo we should fix here, how do you feel about this?
```suggestion
              description="The generated query requires use of MongoDB's aggregation framework. Open and work on it in the Aggregation Pipeline Builder"
```
or with the typo fixed:
```suggestion
              description="Your query requires stages from MongoDB's aggregation framework. Continue to work on it in our Aggregation Pipeline Builder"
```

File: packages/compass-query-bar/src/stores/ai-query-reducer.ts
Comment (Anemy) [inline packages/compass-query-bar/src/stores/ai-query-reducer.ts:270]: Have we checked with Julia that an `AI Response Failed` event is expected here? Since it's making a usable pipeline it might not be intuitive as a failure.

**Review by Anemy — COMMENTED** [review]
Looking good, two small comments

File: packages/compass-components/src/components/generative-ai/generative-ai-input.tsx
Comment (alenakhineika) [reply packages/compass-components/src/components/generative-ai/generative-ai-input.tsx:265]: I think "Open and work on it in the Aggregation Pipeline Builder" is a bit confusing because this message shows up as a guide cue when we already navigated to the aggregation. But I will fix the typo, thx! did not notice it 👀

File: packages/compass-query-bar/src/stores/ai-query-reducer.ts
Comment (alenakhineika) [reply packages/compass-query-bar/src/stores/ai-query-reducer.ts:270]: Good point. I will be on vacation next week. Can you chat about this with Julia and either adjust this PR or open the follow-up PR if this will be merged?

Comment (Anemy) [reply packages/compass-query-bar/src/stores/ai-query-reducer.ts:270]: Sure thing, I'll ping her now. Not a blocker.

File: packages/compass-components/src/components/generative-ai/generative-ai-input.tsx
Comment (Anemy) [reply packages/compass-components/src/components/generative-ai/generative-ai-input.tsx]: Ah gotcha, I was thinking this was before the transition. Sounds good.

**Comment by gribnoysup** [comment]
@Anemy while Alena is on vacation, I'm making sure we merge this, I did a bit of code clean up (no behavior changes) on top of last changes, do you mind giving this another look and then I'll merge if this looks good to you?

File: packages/compass-query-bar/src/stores/ai-query-reducer.ts
Comment (Anemy) [reply packages/compass-query-bar/src/stores/ai-query-reducer.ts:270]: Got lgtm here: https://mongodb.slack.com/archives/C05AH835WQN/p1692996996219599?thread_ts=1692738027.149129&cid=C05AH835WQN

File: packages/compass-components/src/components/generative-ai/generative-ai-input.tsx
Comment (Anemy) [inline packages/compass-components/src/components/generative-ai/generative-ai-input.tsx:154]: A nit, fine as is, if we have the value in the store not as possibly undefined I prefer to keep the component interfaces the same.
```suggestion
  isAggregationGeneratedFromQuery: boolean;
```
That way if something changes in the store we're for sure to catch it here. In tests we can then default it to `false`.

**Review by Anemy — APPROVED** [review]
lgtm!

**Pull request merged**
